From bb9b97b8cd99c875a836ce782121e5b8e879baea Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sun, 6 Jul 2025 20:15:05 +0200
Subject: [PATCH 060/115] wip: phy: mediatek: tphy: Support phone products

Add support for phone products in the MediaTek TPHY driver by setting
force bits on init.

Signed-off-by: Dinolek <git@dinolek.me>
---
 drivers/phy/mediatek/phy-mtk-tphy.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/phy/mediatek/phy-mtk-tphy.c b/drivers/phy/mediatek/phy-mtk-tphy.c
index acf506529..5d9bce404 100644
--- a/drivers/phy/mediatek/phy-mtk-tphy.c
+++ b/drivers/phy/mediatek/phy-mtk-tphy.c
@@ -111,7 +111,12 @@
 
 #define U3P_U2PHYDTM1		0x06C
 #define P2C_RG_UART_EN			BIT(16)
+#define P2C_FORCE_VBUSVALID		BIT(13)
+#define P2C_FORCE_SESSEND		BIT(12)
+#define P2C_FORCE_BVALID		BIT(11)
+#define P2C_FORCE_AVALID		BIT(10)
 #define P2C_FORCE_IDDIG		BIT(9)
+#define P2C_FORCE_IDPULLUP		BIT(8)
 #define P2C_RG_VBUSVALID		BIT(5)
 #define P2C_RG_SESSEND			BIT(4)
 #define P2C_RG_AVALID			BIT(2)
@@ -861,6 +866,12 @@ static void u2_phy_instance_init(struct mtk_tphy *tphy,
 	/* Workaround only for mt8195, HW fix it for others (V3) */
 	u2_phy_pll_26m_set(tphy, instance);
 
+	/* Used by phone products */
+	mtk_phy_set_bits(com + U3P_U2PHYDTM1,
+			 P2C_FORCE_IDPULLUP | P2C_FORCE_AVALID |
+				 P2C_FORCE_BVALID | P2C_FORCE_SESSEND |
+				 P2C_FORCE_VBUSVALID);
+
 	dev_dbg(tphy->dev, "%s(%d)\n", __func__, index);
 }
 
-- 
2.56.0

